include/envoy/upstream/upstream.h

```diff
@@ -377,6 +377,8 @@ class ClusterInfo {
    */
   virtual bool addedViaApi() const PURE;
 
+  virtual bool addedLazily() const PURE;
```

Comment:
More so than my comment on https://github.com/envoyproxy/data-plane-api/pull/524, I think this is where we might want to think about how to reconcile incremental and lazy loading. As a strawman, I'd suggest there is no difference between an incremental cluster addition on a regular CDS channel and a lazy loaded cluster. WDYT? @mattklein123 as well.

Reply:
I agree. Ideally if we could extend the xDS API with incremental updates and incremental subscriptions (or hint) we could handle lazy loading with a single stream and reconcile quite a bit of this logic.

Has there been any work / discussion on this? Should I start a data-plane-api proposal?

Reply:
I started a strawman proposal here: https://github.com/envoyproxy/data-plane-api/pull/527

Reply:
I need to spend heads down time with this next week, but at a high level if we can handle lazy loading w/ incremental lets just do incremental. We will get much bigger bang for the buck here. I will focus mostly on helping work through the issues in envoyproxy/data-plane-api#527.